Update Rev Links and add SPARK Flex (#2527)
wpilibsuite/frc-docs: source/docs/software/can-devices/third-party-devices.rst

REV Robotics
------------

> REV Robotics currently offers the SPARK MAX motor controller which can be used for brushed and brushless (Rev NEO and Rev NEO 550) motors as well as the Spark Flex motor controller.

Reviewer: ```suggestion
REV Robotics currently offers the SPARK MAX and SPARK Flex motor controllers which can be used for brushed and REV brushless (NEO, NEO 550, and NEO Vortex) motors.
```

REV Motor Controllers
^^^^^^^^^^^^^^^^^^^^^

- **SPARK MAX**

>     - API Documentation (`Java <https://codedocs.revrobotics.com/java/com/revrobotics/package-summary.html>`__, `C++ <https://codedocs.revrobotics.com/cpp/namespacerev.html>`__)

Reviewer: ```suggestion
    - API Documentation (`Java <https://codedocs.revrobotics.com/java/com/revrobotics/cansparkmax>`__, `C++ <https://codedocs.revrobotics.com/cpp/classrev_1_1_c_a_n_spark_max>`__)
```

    - `Technical Manual <https://docs.revrobotics.com/brushless/spark-max/overview>`__

- **SPARK Flex**

>     - API Documentation (`Java <https://codedocs.revrobotics.com/java/com/revrobotics/package-summary.html>`__, `C++ <https://codedocs.revrobotics.com/cpp/namespacerev.html>`__)

Reviewer: ```suggestion
    - API Documentation (`Java <https://codedocs.revrobotics.com/java/com/revrobotics/cansparkflex>`__, `C++ <https://codedocs.revrobotics.com/cpp/classrev_1_1_c_a_n_spark_flex.html>`__)
```
